### PR TITLE
Add support for non-Latin paths on Windows

### PR DESCRIPTION
**Pull request by rfomin**
I did not change the `data/rd_*` files, and there is `fopen()` under the `#TEST` macros in `midifile.c`.

Alternative to #204

### CODE REVIEW

File: prboom2/src/m_file.c [hunk 1]
```diff
@@ -65,22 +70,238 @@
...
+  wchar_t *wstr = NULL;
+
+  wstr = ConvertSysNativeMBToWide(str);
+
+  if (!wstr)
+    return NULL;
+
+  ret = ConvertWideToUtf8(wstr);
+
+  free(wstr);
```
Comment (kraflab) [inline prboom2/src/m_file.c]: Could you use zone functions for all the frees and allocations?

Comment (rfomin) [reply prboom2/src/m_file.c]: Then we also need to use the zone memory in `i_glob.c` Is this necessary?

Comment (kraflab) [reply prboom2/src/m_file.c]: Yes, otherwise we'll be using mixed memory management, which sounds like a recipe for future bugs 😄

Comment (rfomin) [reply prboom2/src/m_file.c]: Fixed in https://github.com/kraflab/dsda-doom/pull/234/commits/db2a68cf318320ce2ab36805d8645cf3040967d9

**Comment by kraflab** [comment]
This is awesome stuff, thanks! 🥳

**Pull request merged**